### ISSUE TITLE
Allow Exceptions to take std::string as input

### ISSUE THREAD

**Issue by apavlo** [cmu-db/terrier#596]
Our `Exception` class currently only supports `const char *` for the error message:

https://github.com/cmu-db/terrier/blob/master/src/include/common/exception.h#L39-L40

But it is easier to construct error messages with `std::string` with embedded information. This is causing some people to explicitly convert `std::string` in their code:

https://github.com/cmu-db/terrier/pull/475/files#diff-b4603b98bf01929a9ba25807c768fe0bR213

We should just add a new constructor to `Exception`:

```c++
Exception(const ExceptionType type, const std::string &msg, const char *file, int line) :
      : Exception(type, msg.c_str(), file, line) {}
```

The only potential issue with this is that we may want to have non-dynamic error messages and therefore we should not be passing in a `std::string`

**Comment by achanana** [comment]
> we may want to have non-dynamic error messages

Is there a way we can completely avoid dynamic error messages? If we are embedding information we would need that right?

**Comment by eppingere** [comment]
This was resolved in #959
